designer/fitting/dwipi.py

```diff
@@ -5,13 +5,17 @@
 import os
 import random as rnd
 import warnings
+
```

Comment:
No need to add a blank line here

Reply:
> Extra blank lines should be deleted, and I'm concerned about a function definition in a function definition.

I have another PR coming in to clean up and remove unused functions. That PR will take care of whitespaces and formatting.